### PR TITLE
Adds Network Status Callback registration page

### PR DESCRIPTION
**Pull request by VeijoPesonen**
Replaces already existing documentation about the subject. Explains
a bit further what is required from the application POV. @SeppoTakalo, @jarlamsa have a look and please let me know how to improve it. @AnotherButler Is it somehow possible to render the documentation(this PR) so I would see that all the links are working etc?

### CODE REVIEW

**Review by jarlamsa — CHANGES_REQUESTED** [review]

File: docs/reference/api/connectivity/networksocket/NetworkStatus.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            break;
+        default:
+            printf("Not supported");
+            break;
+    }
+}
+```
+
+Now, the callback function is to be registered to the interface. The socket needs to be set in
+non-blocking mode for the asynchronous behavior.
```
[the remark this answers is not in the document]
Comment (SeppoTakalo) [reply docs/reference/api/connectivity/networksocket/NetworkStatus.md]: NetworkInteface MAY be set to non-blocking mode to achieve asynchronous behaviour. But it is not forced.

It only blocks the first `connect()` call. Then disconnection and others will cause callback to be triggered.

So `set_blocking()` only affects the `connect()` behaviour.

Can you clarify those?

**Review by SeppoTakalo — CHANGES_REQUESTED** [review]
Add this to the `docs.json` as well.

**Comment by SeppoTakalo** [comment]
Other that those that I commented, It looks good.

**Review by SeppoTakalo — APPROVED** [review]
Update the documentation. This is now good to go from my point of view.

File: docs/reference/api/connectivity/networksocket/NetworkStatus.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+Optionally, the application might want to set the `connect()` method to nonblocking mode and wait until connectivity is fully established.
+
+```cpp
+    eth.set_blocking(false);
+    eth.connect();              // Return immediately
+```
+
+By default, the `connect()` call blocks until `NSAPI_STATUS_GLOBAL_UP` state is reached. Some applications might require only link-ocal connectivity and therefore do not need to block that long. In those case monitoring the state changes is the preferred behavior.
+
```
Comment (AnotherButler) [inline docs/reference/api/connectivity/networksocket/NetworkStatus.md]: Query: Could you please link to the class reference in the Doxygen? I think it should be in Data Structures here, but I don't see it. Could it be somewhere else?

https://os-doc-builder.test.mbed.com/docs/development/mbed-os-api-doxy/annotated.html

**Review by AnotherButler — COMMENTED** [review]
@VeijoPesonen Thanks for the PR 👍 Please review my copy edits to make sure I didn't accidentally change the meaning of anything, and please address my comment.

**Comment by AnotherButler** [comment]
@jarlamsa Are your concerns addressed?

File: docs/reference/api/connectivity/networksocket/NetworkStatus.md (hunk 2 above)
Comment (SeppoTakalo) [reply docs/reference/api/connectivity/networksocket/NetworkStatus.md]: I cannot find it either..
However, I don't see much benefit on Doxygen linking as it only need this `nsapi_connection_status_t` which is basically already shown here.

**Comment by SeppoTakalo** [comment]
jarlamsa is travelling this week, but I would assume that yes, concerns have been addressed, as those we same as mine.
I did change the documentation to fix these.

**Comment by SeppoTakalo** [comment]
Fixed conflicts.
@AnotherButler This is now ready to go in.
Jarno is travelling this week, so we cannot wait him to click "Approve".

**Comment by javiermelendezc** [comment]
Documentation on connect() and disconnect() says: 

> This blocks until connection is established, but asynchronous operation can be enabled by calling NetworkInterface::set_blocking(false).

Then in set_blocking() parameters description it says:

> virtual nsapi_error_t set_blocking(bool  blocking)	
> ....
> blocking | Use true to set NetworkInterface in asynchronous mode.

I think it should be: **Use false to set NetworkInterface in asynchronous mode.**

**Pull request merged**